scimma_admin/hopskotch_auth/models.py

```diff
@@ -507,3 +507,30 @@ def remove_user_group_permissions(user_id, group_id):
 def delete_topic_permissions(topic_id):
     CredentialKafkaPermission.objects.filter(topic=topic_id).delete()
     GroupKafkaPermission.objects.filter(topic=topic_id).delete()
+
+# locate all topics which the user has permission to access in some way, either because:
+# 1. they are publicly readable
+# 2. the user belongs to a group which has some permission providing access
+# The user may not currently have any credential configured to use the allowed access in the latter
+# case, however.
+# Returns a dictionary mapping topic names to a strings describing the nature of the access. 
+def topicsAccessibleToUser(user_id):
+    accessible = {}
+
+    # first, handle access via group permissions
+    user_memberships = GroupMembership.objects.filter(user=user_id)
+    for membership in user_memberships:
+        group = membership.group
+        group_permissions = GroupKafkaPermission.objects.filter(principal=group).select_related('topic')
+        for permission in group_permissions:
+            accessible[permission.topic.name] = "via group permission"
```

Comment:
Might be nice to say which group grants permission? Seems easy to do here, too.

Reply:
I had not done this before because a user may have access via more than one group, and it seemed like a pain to try to find which group gve broader access (if any did). These seem like rare issues, though, so just reporting any relevant group is probably useful for most users, and that change is indeed very simple.